cmake/external/neural_speed.cmake

```diff
@@ -0,0 +1,18 @@
+set(NEURAL_SPEED_URL https://github.com/intel/neural-speed.git)
+set(NEURAL_SPEED_TAG 18720b319d6921c28e59cc9e003e50cee9a85fcc) # kernel-only release v0.2
+
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" AND onnxruntime_target_platform STREQUAL "x86_64")
+  set(USE_NEURAL_SPEED TRUE)
+elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC" AND onnxruntime_target_platform STREQUAL "x64")
+  set(USE_NEURAL_SPEED TRUE)
+endif()
+
+if(USE_NEURAL_SPEED)
+  FetchContent_Declare(
+      neural_speed
+      GIT_REPOSITORY ${NEURAL_SPEED_URL}
+      GIT_TAG        ${NEURAL_SPEED_TAG}
+  )
+  FetchContent_MakeAvailable(neural_speed)
+  add_compile_definitions(ORT_NEURAL_SPEED)
```

Comment:
Who uses this marco?

Reply:
It is used by onnxruntime\contrib_ops\cpu\quantization\matmul_nbits.cc and onnxruntime\test\contrib_ops\matmul_4bits_test.cc

Reply:
Then we do not need to appy it to external projects such as googletest/abseil/protobuf.  Please move the code to https://github.com/microsoft/onnxruntime/blob/main/cmake/CMakeLists.txt#L867